e2e/ctl_v3_test.go

```diff
@@ -155,27 +168,61 @@ func testCtl(t *testing.T, testFunc func(ctlCtx), opts ...ctlOption) {
 	}
 }
 
+var envMap = make(map[string]struct{})
+
+func init() {
+	// Keep up-to-date with 'prefixArgs'
+	fs := []string{
+		"endpoints",
+		"dial-timeout",
+		"insecure-transport",
+		"insecure-skip-tls-verify",
+		"cacert",
+		"cert",
+		"key",
+		"user",
+	}
+	for _, f := range fs {
+		envMap[flags.FlagToEnv("ETCDCTL", f)] = struct{}{}
+	}
+}
```

Comment:
@xiang90 Ended up having global `envMap`, because we pass `ctlCtx` by value (not pointer). To get `fmap` field updated, we need to change all e2e test function argument types to pointer.

Or have this one common `envMap` to unset after tests, which is simpler.

Another concern is if several tests with `flagByEnv` true were running at the same time, it could affect other ongoing tests. I don't mind dropping these tests. What do you think?

Thanks.